determined-ai/determined
# chore: Use UI Kit Button in application [WEB-771]

## Description

chore: Use UI Kit Button in application WEB-771

Updates all usages of `<button/>` and Antd `<Button>` to use the new UI component. Except in one instance which is discussed in (1) in the `Additional Commentary` section. 

A `htmlType` has been added to the UIKit `Button` this allows us to specify a button with as a `submit` type.




## Test Plan

1. Most pages in the WebUI have been updated with this PR. Click through the application and ensure that visually nothing looks broken. Makes sure to check the UI in smaller screen sizes also. 

2.  Check these specific pages and ensure all buttons functionality looks correct:

- Sign in page  
- The action dropdown in tables 
- The gallery modal. This can be reached by clicking on a graph on the `HP Heat Map` page in an experiment.  Double check that links work and look correct
-  The Metadata field in the `Create Model` modal 
- The navigation sidebar both full size and collapsed.
-  The foldable experiment and trial page headers (can be accessed from clicking the dropdown icon in the same row as the experiment/trial id and name on the overview page) 





## Commentary (optional)

1. The only usage of the `Antd` Button left are the [custom animated buttons](https://github.com/determined-ai/determined/blob/be67810f277873e6abc26f7c625631ca0dba6a25/webui/react/src/pages/ExperimentDetails/ExperimentDetailsHeader.tsx#L521) that show experiment status. My assumption is that these should count as a new type of button in the UI kit? 

2.  The `ModelVersionHeader` behavior was a bit confusing to me. Currently it shows a small blank list because of a `.buttonAction` css class that was sit to not display anything, look at the top right of the following image beneath the "dropdown" icon.

![Screen Shot 2023-01-12 at 11 09 27 AM](https://user-images.githubusercontent.com/103522725/212133512-8388f542-40a7-4911-a2e8-cfcd34d8c452.png)

With this update
![Screen Shot 2023-01-17 at 4 51 02 PM](https://user-images.githubusercontent.com/103522725/213029496-99335c45-3219-4c1d-9ab3-d3cdcc0c340a.png)

te the dropdown does show an option.

My guess would be the updated behavior is what we want, however I did want to call this out. 

3. When a user cannot create a new workspace we show a disabled button with a title. The is an `override` [css class](https://github.com/determined-ai/determined/pull/5726/files#diff-928beaeb8789a47b364cf46b21ba7ab5f453683877aac2bc02fdf47ea6697bbfL342) that looks like it tries to still show a button cursor but it does not work, hovering over the button does not show a cursor. Additionally, we show a `title` on the button, but that has been removed. My initial thought is we should instead be using a tooltip to show the message to the user instead?



## Checklist

~- [ ] User-facing API changes need the "User-facing API Change" label.~
~- [ ] Release notes should be added as a separate file under `docs/release-notes/`.~
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
~- [ ] Licenses should be included for new code which was copied and/or modified from any external code.~

## Ticket
WEB-771
